How to adapt to typing change for always_iterable?

Over in setuptools I noticed that the typing tests started failing with more-itertools 11.0:

```
setuptools/config/expand.py:125: error: Argument 2 to "join" has incompatible type "PathLike[str] | Iterable[str | PathLike[str]]"; expected "str | PathLike[str]"  [arg-type]
```

See e.g. [this CI run](https://github.com/pypa/setuptools/actions/runs/23911615010/job/69734806337?pr=5207). If I pin more-itertools to `<11.0`, the issue goes away.

I asked my resident typing expert @jorenham, and he thinks the issue might be this code:

https://github.com/more-itertools/more-itertools/blob/5ea81c48727f9906879e30b352772bc3869c4e88/more_itertools/more.pyi#L444-L447

> This looks like the issue: The obj: _T will also match obj: Iterable[_T] , so I can see why mypy would infer -> Iterator[_T] | _T because of the overlap

Hmm, here's how `setuptools` looks with `more_itertools` at 11.0.1:
```
% mypy setuptools/config/expand.py
setuptools/config/expand.py:125: error: Argument 2 to "join" has incompatible type "PathLike[str] | Iterable[str | PathLike[str]]"; expected "str | PathLike[str]"  [arg-type]
setuptools/config/_apply_pyprojecttoml.py:336: error: Cannot access instance-only attribute "global_options" on class object  
```

With the proposed fix in #1146:
```
% mypy setuptools/config/expand.py
setuptools/config/expand.py:125: error: No overload variant of "join" matches argument types "str", "object"  [call-overload]
setuptools/config/expand.py:125: note: Possible overload variants:
setuptools/config/expand.py:125: note:     def join(str, /, *paths: str) -> str
setuptools/config/expand.py:125: note:     def join(str | PathLike[str], /, *paths: str | PathLike[str]) -> str
setuptools/config/expand.py:125: note:     def join(bytes | PathLike[bytes], /, *paths: bytes | PathLike[bytes]) -> bytes
setuptools/config/_apply_pyprojecttoml.py:336: error: Cannot access instance-only attribute "global_options" on class object  
```

With the proposed fix in #1144:
```
% mypy setuptools/config/expand.py
setuptools/config/_apply_pyprojecttoml.py:336: error: Cannot access instance-only attribute "global_options" on class object  
```

So I think #1144 is the leading candidate at the moment.

--- a/more_itertools/more.pyi
+++ b/more_itertools/more.pyi
@@ -2,6 +2,7 @@
 
 from __future__ import annotations
 
+import sys
 import types
 
 from collections.abc import (
@@ -170,7 +171,11 @@ _GenFn = TypeVar('_GenFn', bound=Callable[..., Iterator[Any]])
 _Raisable = BaseException | type[BaseException]
 _NumberT = TypeVar("_NumberT", float, Decimal, Fraction)
 
-_ClassInfo = type | types.UnionType | tuple[_ClassInfo, ...]
+# The type of isinstance's second argument (from typeshed builtins)
+if sys.version_info >= (3, 10):
+    _ClassInfo = type | types.UnionType | tuple[_ClassInfo, ...]
+else:
+    _ClassInfo = type | tuple[_ClassInfo, ...]
 
 @type_check_only
 class _SizedIterable(Protocol[_T_co], Sized, Iterable[_T_co]): ...
@@ -435,30 +440,6 @@ def sort_together(
 ) -> list[tuple[_T, ...]]: ...
 def unzip(iterable: Iterable[Sequence[_T]]) -> tuple[Iterator[_T], ...]: ...
 def divide(n: int, iterable: Iterable[_T]) -> list[Iterator[_T]]: ...
-@overload
-def always_iterable(
-    obj: None, base_type: Any = ...
-) -> Iterator[tuple[()]]: ...
-@overload
-def always_iterable(obj: bytes, base_type: None) -> Iterator[int]: ...
-@overload
-def always_iterable(obj: Iterable[_T], base_type: None) -> Iterator[_T]: ...
-@overload
-def always_iterable(obj: _T, base_type: None) -> Iterator[_T]: ...
-@overload
-def always_iterable(
-    obj: bytes, base_type: type[bytes] | tuple[type[str], type[bytes]] = ...
-) -> Iterator[bytes]: ...
-@overload
-def always_iterable(
-    obj: Iterable[_T],
-    base_type: type[bytes] | type[str] | tuple[type[str], type[bytes]] = ...,
-) -> Iterator[_T]: ...
-@overload
-def always_iterable(
-    obj: _T, base_type: tuple[type[str], type[bytes]] = ...
-) -> Iterator[_T]: ...
-@overload
 def always_iterable(
     obj: object,
     base_type: _ClassInfo | None = ...,

